checks.d/consul.py

```diff
@@ -102,7 +124,18 @@ def _is_instance_leader(self, instance):
             return False
 
     def _check_for_leader_change(self, instance):
-        agent_dc = self._get_agent_datacenter(instance)
+        perform_new_leader_checks = instance.get('new_leader_checks',
+                                                 self.init_config.get('new_leader_checks', False))
+        perform_self_leader_check = instance.get('self_leader_check',
+                                                 self.init_config.get('self_leader_check', False))
+
+        if perform_new_leader_checks and perform_self_leader_check:
```

Comment:
Yeah, this was all an attempt to preserve the existing behavior as-is. There would be other options, but it wasn't immediately clear how I'd preserve the old behavior for people who might be relying on it.